### ISSUE TITLE
MixedArgument "Argument cannot be mixed" wrongly reported for typed constants in PHP 8.3

### ISSUE THREAD

**Issue by martinjoiner** [vimeo/psalm#10700]
PHP 8.3 introduced typed constants. In the following example the `PAGE_SIZE` constant is hard-typed to be an integer. 

```php
class APIFetchService 
{
    public const int PAGE_SIZE = 20;
}
```

However, if this constant is used as an argument on a method that expects an integer as the example below, Psalm will return `info` level warning https://psalm.dev/030 

```php

function takesInt(int $i) : void {}

takesInt(APIFetchService::PAGE_SIZE);

```

Happy to contribute effort to fixing this. I've not got the Psalm project set up ready to dev on my local machine but happy to write unit test examples and/or manually test dev branches.
